### DISCUSSION TITLE
Why do I find this confusing...

### DISCUSSION THREAD

**Discussion by microfoundry** [mathieucarbou/MycilaESPConnect#31] [General]
Greetings Mathieu - I'm working with your library and I'm trying to understand the flow and why this this is written as such:

src/MycilaESPConnect.cpp Line 385:
```
  if (_state == Mycila::ESPConnect::State::AP_STARTED || _state == Mycila::ESPConnect::State::NETWORK_CONNECTED) {
    _disableCaptivePortal();
  }

```

Maybe I'm missing the bigger picture...

-Terry

**Comment by mathieucarbou** [comment]
hello!

when in captive portal mode, when the user select its choice (either ap mode or wifi), we activate AP mode (or try connect to wifi with timeout). Once done, captive portal is disabled.

usually, this is not something observed because most of the case, after a use choice is done, ESPConnect will reboot the ESP (there is a reboot option set to true by default). This is often easier to just reboot and restart with the proper network settings.

But if a reboot is not possible, option can be set to false, ESPConnect will disable captive portal and connect to network or AP mode. But you need to be careful with that, especially when using handlers, because network interface changes and ESPAsyncWS handlers should contain some filter rules depending on the state of the captive portal.

**Comment by microfoundry** [comment]
Ok - thank you for the clarification. With your explanation and another read through the code, the AP vs STA modes make more sense...

I appreciate it!
